### ISSUE TITLE
After clicking "Check for updates" in Subscriptions, it "sometimes" showed the wrong time of clicking.

### ISSUE THREAD

**Issue by lancelot-moon** [AdguardTeam/AdguardForiOS#635]
In normal situation, it should show the correct time (When did we click "Check for updates"?).
But it "sometimes" shows the earlier time which we clicked "Check for updates" before.

This is the abnormal situation. (No subscription was updated.)
The last time of clicking "Check for updates" in Subscriptions was 22:28 in my video.
<details><summary>Screenshot:</summary>

![image](https://i.imgur.com/aVY3vk2.png)
</details><br/>

At 22:32, I again clicked "Check for updates".
https://youtu.be/rnvWxTmETbc

After checking, it should show that the time was 22:32.
But it still showed 22:28 finally.

This is the normal situation. (No subscription was updated.)
https://youtu.be/2M23fFpSNDk

The last time of clicking "Check for updates" in Subscriptions was 22:25 in the above video.
At 22:27, I again clicked "Check for updates".
After checking, it correctly showed 22:27 finally.

I have 90497 rules in Subscriptions. (I'm trying its accurate upper limit.)
The VPN tunnel is still normally working.
About the wrong time of clicking "Check for updates" in Subscriptions, I don't know if it's because my rules approach its upper limit (around 91k rules).

The abnormal situation SOMETIMES occurred.
You could try more times if it also happened in the condition of only a few rules.

AdGuard v2.0.0 beta version indeed still has some unsolved bugs.
And then please don't forget to integrate Web Reporting Tool in AdGuard v2.0.0 release version according to your original plan.
We translators completed the translations of Web Reporting Tool (localization).
It's an important feature which maybe helps non-English speaking users to more actively report issues.
( Great, I just went to check that it's done.
Maybe I can try it in the next beta version and check for my translations.
Thanks for your hard working! )

B/R

**Comment by Sailor-Chibimoon** [comment]
This is a nitpicking test. Base on statistics (in forums) an average person normally opens AdGuard app twice a week to update the subcriptions.

If this is a bug, i think this is low priority. 

And can be fixed on later date. Probably remove the time completely and just put the date.

Not everyone checks the time. They check the date, but i doubt if they check the time. 

Selected people like @lancelot-moon probably checks for updates multiple times a day so he might find this important.

For me as long as the date is correct, i dont care about the time since I am one of those average people who only open the app to update twice a week.

I’m not having a go at you. This is just based on facts and numbers I read online.

**Comment by ameshkov** [comment]
@IvanIin are we showing the last update check time or the last time when anything actually changed in the subscriptions?

**Comment by lancelot-moon** [comment]
@ameshkov
I supposed that it shows the time of clicking "Check for updates".

In the most situation, it shows the above time even if no filter is updated.

But it sometimes showed the earlier time which I clicked before.

Wait for Ivanlin, he'll tell us how it works.))

As a tester, we testers report critical or small bugs as long as they're bugs.

And I didn't say that this is a critical bug.
Why did someone have such excessive response?

We test AdGuard in any conditions for helping AdGuard team.

It seems that someone still doesn't understand the situation and difference.

Maybe she/he was sleeping in the dream.

We're not general users who only enjoy the result.

We're testers and report any kinds of bugs to AdGuard team.

Ameshkov, I came here for happily discussing about AdGuard with your team and those who like AdGuard.

I came here for letting AdGuard's Taiwanese users have quality Traditional Chinese translations.

Please let unwelcome someone leave me alone.

There are many other posts on the AdGuard GitHub.

Why does someone so like to reply my articles?
I think it's not because that he/she loves me. >///<

Simply, don't bother me, someone.
You have annoyed me.

Thank AdGuard team very much!

**Comment by Sailor-Chibimoon** [comment]
😂😂😂 Wow very sexist! I’m a she, not a he. I’m a beautiful girl, a loving wife and is expecting (8 months preggy) 😊 I tried to be serious earlier but i think you miss the old and jolly me. 

I reply to other people too, you can check, but i like you the most. I really like your user name. I don’t care about other people.

I even get alerted when you post. So cute! 🤩🤩🤩

I will name that to my future kid. 

Don’t be so harsh on me...

**Comment by ameshkov** [comment]
@lancelot-moon 
> I supposed that it shows the time of clicking "Check for updates".

It should do that indeed.

@Sailor-Chibimoon 

Stop that please.

**Comment by Sailor-Chibimoon** [comment]
‘Kay! Sorry 😊😊though you really need to check on him if he is ok. PM him. There was an earthquake in taiwan as per news. It was horrible. Delete this comment if required. ✌🏻✌🏻✌🏻 Sorry again

**Comment by X8716e** [comment]
@lancelot-moon & @ameshkov - I'll piggyback on this issue and raise one myself, as I believe it’s of the same making. Moreover, I belive @lancelot-moon has opened up the surface of the deeper issue at play.

- [x] I’ve not been able to request updates for _any filter list hosted anywhere_. All that occurs is the appearence of Apple’s infamous never-ending spinning wheel. This includes:
- [x] Subscriptions on Github.
- [x] Subscriptions given via the AdGuard Pro application interface - using any 3 sources provided, AdGuard SDN, hpHosts, or AdAway.
- [x] Customizable subscriptions based at the Sans Institute’s [Internet Storm Center](https://isc.sans.edu/).

**Comment by X8716e** [comment]
2.0.0 (132) release - 

- [x] fixed

**Comment by lancelot-moon** [comment]
@IvanIin

No, it still "sometimes" showed the wrong time of clicking in beta v2.0.0(132) after clicking "Check for updates" in Subscriptions.

I don't have sure reproduction steps.
But there are rare chances to see the issue if deleting a subscription, adding a subscription and then checking for updates.

For example:
The time of "Check for updates" and the time of last modified in each subscription showed 7:00 from the beginning.
If we clicked "Check for updates" at 8:00, and the abnormal issue occurred...
The time of "Check for updates" would still show 7:00, not 8:00.
Especially, even if those subscriptions were not updated after clicking "Check for updates"(The numbers of rules were not changed.), some subscriptions would show that their time of last modified are 7:00, and other subscriptions would show that their time of last modified are 8:00.
ex:
1hosts.cf filter and some subscriptions (time of last modified: 7:00)
AdGuard SDN filter and other subscriptions (time of last modified: 8:00)

It's in disorder.

Thanks for your hard working!

**Comment by lancelot-moon** [comment]
@Ivanlin

I just clicked “check for updates” and didn’t do other things.
Finally, I successfully recorded the video.

I clicked “check for updates” at 23:51 on 13th February.
After checking, it still shows the earlier time 2:08 on 12th February.
https://youtu.be/SRwovVDHATw

And then I checked the last modified time in each subscription.
Some subscriptions show 2:07, and others which maybe didn't actually update show 11:51.
It's in disorder!
https://youtu.be/4Ky39RFlAjs
